bake/hcl.go

```diff
@@ -1,41 +1,11 @@
 package bake
 
 import (
-	"strings"
-
 	"github.com/hashicorp/hcl/v2"
-	"github.com/hashicorp/hcl/v2/hclparse"
 	"github.com/moby/buildkit/solver/errdefs"
 	"github.com/moby/buildkit/solver/pb"
 )
 
-func ParseHCLFile(dt []byte, fn string) (*hcl.File, bool, error) {
-	var err error
-	if strings.HasSuffix(fn, ".json") {
-		f, diags := hclparse.NewParser().ParseJSON(dt, fn)
-		if diags.HasErrors() {
-			err = diags
-		}
-		return f, true, err
-	}
-	if strings.HasSuffix(fn, ".hcl") {
-		f, diags := hclparse.NewParser().ParseHCL(dt, fn)
-		if diags.HasErrors() {
-			err = diags
-		}
-		return f, true, err
-	}
-	f, diags := hclparse.NewParser().ParseHCL(dt, fn+".hcl")
```

Comment:
I think in the switch this logic gets dropped? I think we need to keep separate `case "hcl"` and `default` cases in the `switch filepath.Ext(strings.ToLower(f.Name)) {` above.

Reply:
I was wondering why we have this fallback to json if hcl parsing fails. Looks like a best effort if user don't use a known extension. I'll bring it back.